source/transactions-convenient-api/transactions-convenient-api.rst

```diff
@@ -384,21 +376,36 @@ their users to design idempotent callbacks.
 
 .. _Retry Transactions and Commit Operation: https://docs.mongodb.com/manual/core/transactions/#retry-transaction-and-commit-operation
 
-The commit is not retried after a write concern timeout error
--------------------------------------------------------------
+The commit is retried after a write concern timeout error
+---------------------------------------------------------
 
-Per the Transactions specification, drivers internally retry commits that fail
-with an `UnknownTransactionCommitResult`_ error label. This label is applied to
-write concern errors other than UnsatisfiableWriteConcern
-(CannotSatisfyWriteConcern in server versions before 4.2) and
-UnknownReplWriteConcern, and may appear on a WriteConcernFailed error, which
-indicates a write concern timeout.
+Per the Transactions specification, drivers internally retry
+``commitTransaction`` once if it fails due to a retryable error (as defined in
+the `Retryable Writes`_ specification). Beyond that, applications may manually
+retry ``commitTransaction`` if it fails with any error bearing the
+`UnknownTransactionCommitResult`_ error label. This lable is applied for the
+the following errors:
 
-This specification intentionally chooses not to retry commit operations after a
-write concern timeout error, as doing so would exceed the user's original
-intention for ``wtimeout``.
+.. _Retryable Writes: ../retryable-writes/retryable-writes.rst#terms
 
-.. _UnknownTransactionCommitResult: https://github.com/mongodb/specifications/blob/master/source/transactions/transactions.rst#unknowntransactioncommitresult
+.. _UnknownTransactionCommitResult: ../transactions/transactions.rst#unknowntransactioncommitresult
+
+- Server selection failure
+- Retryable error (as defined in the `Retryable Writes`_ specification)
+- Write concern failure or timeout (excluding UnsatisfiableWriteConcern and
+  UnknownReplWriteConcern)
+
+A previous design for ``withTransaction`` retried for all of these errors
+*except* for write concern timeouts, so as not to exceed the user's original
+intention for ``wtimeout``. The current design of this specification no longer
+excludes write concern timeouts, and simply retries ``commitTransaction`` within
+its timeout period for any errors bearing the "UnknownTransactionCommitResult"
+label.
+
+This change was made in light of the forthcoming Client-side Operations Timeout
+specification (see: `Future Work`_), which we expect will allow the current
+120-second timeout for ``withTransaction`` to be customized and also obviate the
+need for users to specify ``wtimeout``.
```

Comment:
IIRC, one of the main concerns was that it'd be difficult to explain to how the `withTranasction` timeout (be it 120-second default now or client-side timeout later) interacted with `wtimeout` and just stick with a single timeout; however, I didn't want to dive too much into that. I think the bit about client-side operation timeout ultimately making `wtimeout` a thing of the past (for explicit application use at least) is sufficient.

Reply:
Thanks for explaining the rationale, makes sense to me.